src/payment/strategies/cybersource/cybersource-payment-strategy.spec.ts

```diff
@@ -0,0 +1,357 @@
+        );
+
+        orderActionCreator = new OrderActionCreator(
+            new OrderRequestSender(createRequestSender()),
+            new CheckoutValidator(new CheckoutRequestSender(createRequestSender()))
+        );
+
+        paymentActionCreator = new PaymentActionCreator(
+            new PaymentRequestSender(createPaymentClient()),
+            orderActionCreator
+        );
+
+        strategy = new CyberSourcePaymentStrategy(
+            store,
+            paymentMethodActionCreator,
+            orderActionCreator,
+            paymentActionCreator,
+            cardinalClient
+        );
+
+        payload = merge({}, getOrderRequestBody(), {
+            payment: {
+                methodId: paymentMethodMock.id,
+                gatewayId: paymentMethodMock.gateway,
+            },
+        });
+
+        loadPaymentMethodAction = of(createAction(PaymentMethodActionType.LoadPaymentMethodSucceeded, paymentMethodMock, { methodId: paymentMethodMock.id }));
+        initializePaymentAction = of(createAction(RemoteCheckoutActionType.InitializeRemotePaymentRequested));
+        submitOrderAction = of(createAction(OrderActionType.SubmitOrderRequested));
+        submitPaymentAction = of(createAction(PaymentActionType.SubmitPaymentRequested));
+
+        jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod')
+            .mockReturnValue(loadPaymentMethodAction);
+
+        jest.spyOn(remoteCheckoutActionCreator, 'initializePayment')
+            .mockReturnValue(initializePaymentAction);
+
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+
+        jest.spyOn(paymentActionCreator, 'submitPayment')
+            .mockReturnValue(submitPaymentAction);
+
+        jest.spyOn(cardinalClient, 'initialize').mockReturnValue(Promise.resolve());
+    });
+
+    describe('#initialize', () => {
+        beforeEach(async () => {
+            await strategy.initialize({ methodId: paymentMethodMock.id });
+        });
+
+        it('initializes strategy successfully', () => {
+            expect(cardinalClient.initialize).toHaveBeenCalledWith(true);
+        });
+
+        it('throws data missing error when initializing', async () => {
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(undefined);
+
+            try {
+                await strategy.initialize({ methodId: paymentMethodMock.id });
+            } catch (error) {
+                expect(error).toBeInstanceOf(MissingDataError);
+            }
+        });
+    });
+
+    describe('#execute', () => {
+        it('throws error to inform that payment data is missing', async () => {
+            try {
+                await strategy.execute(payload);
+            } catch (error) {
+                expect(error).toBeInstanceOf(MissingDataError);
+            }
+        });
+
+        it('throws data missing error when executing', async () => {
```

Comment:
I guess we are missing when does this happen. When `payment` is missing?

Reply:
I think that will never happen, but `OrderRequestBody` has the `payment` element as optional so we need to validate if it's different to undefined prior to use it